samples/connectivity/bluetooth/companion/src/main/java/com/example/platform/connectivity/bluetooth/ble/CompanionDeviceManagerSample.kt

```diff
@@ -0,0 +1,304 @@
+                    ) {
+                        Text(text = "Find & Associate device")
+                    }
+                    if (errorMessage.isNotBlank()) {
+                        Text(text = errorMessage, color = MaterialTheme.colorScheme.error)
+                    }
+                }
+            }
+        }
+    }
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+private fun getAssociatedDevice(deviceManager: CompanionDeviceManager): AssociatedDevice? {
+    val associatedDevice = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        deviceManager.myAssociations.lastOrNull()?.toAssociatedDevice()
+    } else {
+        // Before Android 34 we can only get the MAC. We could use the BT adapter to find the
+        // device, but to use CDM we only need the MAC.
+        @Suppress("DEPRECATION")
+        deviceManager.associations.lastOrNull()?.run {
+            AssociatedDevice(
+                id = -1,
+                address = this,
+                name = "",
+                device = null,
+            )
+        }
+    }
+    return associatedDevice
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+private fun Intent.getAssociationResult(): AssociatedDevice? {
+    var result: AssociatedDevice? = null
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        result = getParcelableExtra(
+            CompanionDeviceManager.EXTRA_ASSOCIATION,
+            AssociationInfo::class.java,
+        )?.toAssociatedDevice()
+    } else {
+        // Below Android 33 the result returns either a BLE ScanResult, a
+        // Classic BluetoothDevice or a Wifi ScanResult
+        // In our case we are looking for our BLE GATT server so we can cast directly
+        // to the BLE ScanResult
+        @Suppress("DEPRECATION")
+        val scanResult = getParcelableExtra<ScanResult>(CompanionDeviceManager.EXTRA_DEVICE)
+        if (scanResult != null) {
+            result = AssociatedDevice(
+                id = scanResult.advertisingSid,
+                address = scanResult.device.address ?: "N/A",
+                name = scanResult.scanRecord?.deviceName ?: "N/A",
+                device = scanResult.device,
+            )
+        }
+    }
+    return result
+}
+
+@RequiresApi(Build.VERSION_CODES.TIRAMISU)
+private fun AssociationInfo.toAssociatedDevice() = AssociatedDevice(
+    id = id,
+    address = deviceMacAddress?.toOuiString() ?: "N/A",
+    name = displayName?.ifBlank { "N/A" }?.toString() ?: "N/A",
+    device = if (Build.VERSION.SDK_INT >= 34) {
+        associatedDevice?.bleDevice?.device
+    } else {
+        null
+    },
+)
+
+@RequiresApi(Build.VERSION_CODES.O)
+suspend fun requestDeviceAssociation(deviceManager: CompanionDeviceManager): IntentSender {
+    // Match only Bluetooth devices whose service UUID matches this pattern.
+    // For this demo we will match our GATTServerSample
+    val scanFilter = ScanFilter.Builder().setServiceUuid(ParcelUuid(SERVICE_UUID)).build()
```

Comment:
The CI says that this `SERVICE_UUID` cannot be found. It seems that it is in `GATTServerSample.kt`, and I cannot reproduce the error locally.
Luke, can you reproduce this on your end?

Reply:
The branch is super old. I just rebased the branch and i can see the issue. Just fixing it now